russstation/code/modules/clothing/shoes/miscellaneous.dm

```diff
@@ -0,0 +1,45 @@
+/obj/item/clothing/shoes/cowboy/pink
+	name = "pink cowboy boots"
+	icon = 'russstation/icons/obj/clothing/shoes.dmi'
+	mob_overlay_icon = 'russstation/icons/mob/feet.dmi'
+	icon_state = "cowboy_pink"
+
+/obj/item/clothing/shoes/cowboy/clown
+	name = "clownboy boots"
+	icon = 'russstation/icons/obj/clothing/shoes.dmi'
+	mob_overlay_icon = 'russstation/icons/mob/feet.dmi'
+	icon_state = "cowboy_clown"
+	slowdown = SHOES_SLOWDOWN+1
+	pocket_storage_component_path = /datum/component/storage/concrete/pockets/shoes/clown
+	var/enabled_waddle = TRUE
+
+/obj/item/clothing/shoes/cowboy/clown/Initialize()
+	. = ..()
+	AddComponent(/datum/component/squeak, list('sound/effects/clownstep1.ogg'=1,'sound/effects/clownstep2.ogg'=1), 50)
+
+/obj/item/clothing/shoes/cowboy/clown/equipped(mob/user, slot)
+	. = ..()
+	if(slot == ITEM_SLOT_FEET)
+		if(enabled_waddle)
+			user.AddElement(/datum/element/waddling)
+		if(user.mind && user.mind.assigned_role == "Clown")
+			SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "clownboy boots", /datum/mood_event/clownshoes)
+
+/obj/item/clothing/shoes/cowboy/clown/dropped(mob/user)
```

Comment:
Are there ways to remove shoes from someone without this method firing? Because let's say hypothetically acid burning shoes off didn't fire this method, they'd be able to change their shoes, but keep the moodlet and the waddle.